src/Serilog.Enrichers.Sensitive/PathMaskingOperator.cs

```diff
@@ -0,0 +1,52 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+
+namespace Serilog.Enrichers.Sensitive;
+
+/// <summary>
+/// Represents a masking operator for path names.
+/// <remarks>Supports path names for Windows, Linux, and macOS.</remarks>
+/// </summary>
+public class PathMaskingOperator : RegexMaskingOperator
+{
+    private readonly IPathWrapper _pathWrapper;
+    private readonly bool _combineMaskWithPath;
+    private const string PathPattern =
+        @"^(?:[a-zA-Z]\:|\\\\[\w-]+\\[\w-]+\$?|[\/][^\/\0]+)+(\\[^\\/:*?""<>|]*)*(\\?)?$";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PathMaskingOperator"/> class.
+    /// </summary>
+    /// <param name="pathWrapper">The path wrapper.</param>
+    /// <param name="combineMaskWithPath">This means if set to <see langword="true"/> then the mask will combine with the file name or its directory name.</param>
+    public PathMaskingOperator(IPathWrapper pathWrapper, bool combineMaskWithPath = true) : base(PathPattern)
```

Comment:
As the `IPathWrapper` abstraction isn't necessary (see my other comment below) you can simplify this constructor after inlining the `Path.*` methods.

Reply:
Now path wrapper removed